src/main/java/jenkins/branch/MultiBranchProject.java

```diff
@@ -1196,15 +1196,17 @@ private int processHeadUpdate(SCMHeadEvent<?> event, TaskListener global, String
                 }
                 jobs.clear();
                 for (Job i : p.getItems()) {
-                    if (!candidateNames.contains(i.getName())) {
-                        continue;
-                    }
                     if (_factory.isProject(i)) {
                         Branch branch = _factory.getBranch(i);
+                        if (!candidateNames.contains(branch.getName())) {
+                            continue;
+                        }
```

Comment:
This is part of the actual fix. We need to move the check against candidate names to run against the branch name not the job name